lib/python/rose/config_editor/main.py

```diff
@@ -1847,17 +1848,55 @@ def spawn_window(config_directory_path=None, debug_mode=False,
     splash_screen = rose.gtk.splash.SplashScreenProcess(logo, title,
                                                         number_of_events)
     try:
-        MainController(config_directory_path,
-                       load_updater=splash_screen,
-                       load_all_apps=load_all_apps,
-                       load_no_apps=load_no_apps,
-                       metadata_off=metadata_off)
+        mc = MainController(config_directory_path,
+                            load_updater=splash_screen,
+                            load_all_apps=load_all_apps,
+                            load_no_apps=load_no_apps,
+                            metadata_off=metadata_off)
+
     except BaseException as e:
         splash_screen.stop()
         if debug_mode and isinstance(e, Exception):
             # Write out origin information - this is otherwise lost.
             traceback.print_exc()
         raise e
+
+    # open up any initial_namespaces the user has provided us with
+    if initial_namespaces:
+        # if the namespace ends with a / remove it
+        for i in range(len(initial_namespaces)):
+            if len(initial_namespaces[i]) > 1:
+                if initial_namespaces[i][-1] == '/':
+                    initial_namespaces[i] = initial_namespaces[i][0:-1]
+
+        # for each partial namespace get the full namespace
+        full_namespaces = []
+        for ns in mc.data.namespace_meta_lookup:
+            for namespace in initial_namespaces:
+                index = ns.find(namespace)
+                if index > 0:
+                    before = ns[0:(index + len(namespace))]
+                    after = ns[(index + len(namespace)):].split('/')[0]
+                    full_namespaces.append(before + after)
+                    initial_namespaces.remove(namespace)
+                    break
+
+        # open each namespace in a new tab
+        if full_namespaces:
+            for namespace in full_namespaces:
+                # if the namespace begins with a / remove it
+                if namespace[0] == '/':
+                    namespace = namespace[1:]
+                # open namespace
+                try:
+                    mc.view_page(namespace)
+                except:
+                    print 'could not open' + namespace
```

Comment:
Is this a debug statement? Should use a dialog box to report to user and/or write to STDERR.